scripts/dts/gen_defines.py

```diff
@@ -599,6 +603,44 @@ def write_clocks(node):
                 controller.props["clock-frequency"].val)
 
 
+def write_dmas(node):
+    if "dmas" not in node.props:
+        return
+
+    write_phandle_val_list(node.props["dmas"])
+
+    # Client parameter syntax may vary with the DMA controller changes.
+    for dma_i, dma in enumerate(node.props["dmas"].val):
+        if "st,stm32-dma" in dma.controller.compats:
```

Comment:
Could do this instead, to make it clearer that this is the only DMA controller that generates extra #defines:

```python
# Only this controller currently generates extra #defines
if "st,stm32-dma" not in dma.controller.compats:
    continue
```

A helper function could be used too.

Reply:
Ok, thanks